src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobStats.scala

```diff
@@ -6,70 +6,112 @@ import java.util.logging.{Level, Logger}
-                attempt: java.lang.Integer
-              ))
-            case job: DependencyBasedJob =>
-              val query =
-                s"INSERT INTO ${config.cassandraTable()} (id, ts, job_name, job_owner, job_parents, task_state, slave_id, attempt) VALUES (?, ?, ?, ?, ?, ?, ?, ?) USING TTL ${config.cassandraTtl()}"
-              val prepared = statements.getOrElseUpdate(query, {
-                session.prepare(
-                  new SimpleStatement(query).setConsistencyLevel(ConsistencyLevel.valueOf(config.cassandraConsistency())).asInstanceOf[RegularStatement]
-                )
-              })
-              val parentSet: java.util.Set[String] = job.parents.asJava
-              session.executeAsync(prepared.bind(
-                taskStatus.getTaskId.getValue,
-                new java.util.Date(),
-                job.name,
-                job.owner,
-                parentSet,
-                taskStatus.getState.toString,
-                taskStatus.getSlaveId.getValue,
-                attempt: java.lang.Integer
-              ))
-          }
-        case None =>
+  def getJobState(jobName: String) : CurrentState.Value = {
+    /**
+     * NOTE: currently everything stored in memory, look into moving
+     * this to Cassandra. ZK is not an option cause serializers and
+     * deserializers need to be written. Need a good solution, potentially
+     * lots of writes and very few reads (only on failover)
+     */
+    val status = jobStates.get(jobName) match {
+      case Some(s) =>
+        s
+      case _ =>
+        CurrentState.idle
+    }
+    status
+  }
+
+  def updateJobState(jobName: String, state: CurrentState.Value) {
+    var shouldUpdate = true
+    jobStates.get(jobName) match {
+      case Some(s: CurrentState.Value) => {
+        if ((s == CurrentState.running) &&
+            (state == CurrentState.queued)) {
+          //don't update status if already running
+          shouldUpdate = false
+        }
       }
-    } catch {
-      case e: NoHostAvailableException =>
-        resetSession()
-        log.log(Level.WARNING, "No hosts were available, will retry next time.", e)
-      case e: QueryExecutionException =>
-        log.log(Level.WARNING, "Query execution failed:", e)
-      case e: QueryValidationException =>
-        log.log(Level.WARNING, "Query validation failed:", e)
+      case None =>
+    }
+
+    if (shouldUpdate) {
+      log.info("Updating state for job (%s) to %s".format(jobName, state))
+      jobStates.put(jobName, state)
+    }
+  }
+
+  def removeJobState(job: BaseJob) {
+    jobStates.remove(job.name)
+  }
+
+  def jobQueued(job: BaseJob, attempt: Int) {
+    updateJobState(job.name, CurrentState.queued)
+  }
+
+  def jobStarted(job: BaseJob, taskStatus: TaskStatus, attempt: Int) {
+    updateJobState(job.name, CurrentState.running)
+    job match {
+      case job: ScheduleBasedJob =>
+        insertToStatTable(
+            id=Some(taskStatus.getTaskId.getValue),
```

Comment:
Could you apply a little more DRY here? A lot of these fields are common between the 2 job types. You only need to get the non-common fields first, then make one call to `insertToStatTable`.